### PR TITLE
[fmt] fix format conflict

### LINKED ISSUE #25109: [fmt] Build error
**Host Environment**
- OS: [windows 10]
- Compiler: revision

**To Reproduce**
`./vcpkg install fmt`

**Failure logs**
C:\vcpkg>vcpkg install fmt
Computing installation plan...
The following packages will be built and installed:
    fmt[core]:x86-windows -> 8.1.1#1
Detecting compiler hash for triplet x86-windows...
Restored 0 packages from C:\Users\User\AppData\Local\vcpkg\archives in 219.4 us. Use --debug to see more details.
Installing 1/1 fmt:x86-windows...
Building fmt[core]:x86-windows...
-- Using cached fmtlib-fmt-8.1.1.tar.gz.
-- Cleaning sources at C:/vcpkg/buildtrees/fmt/src/8.1.1-11f8359597.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/fmtlib-fmt-8.1.1.tar.gz
-- Applying patch fix-write-batch.patch
-- Applying patch fix-invalid-command.patch
-- Using source at C:/vcpkg/buildtrees/fmt/src/8.1.1-11f8359597.clean
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j9
    Working Directory: C:/vcpkg/buildtrees/fmt/x86-windows-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\fmt\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:112 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/fmt/portfile.cmake:20 (vcpkg_cmake_install)
  scripts/ports.cmake:146 (include)

error: building fmt:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+fmt
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[fmt]+Build+error
Include '[fmt] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-05-05-67e17c1782801cf481be9ac0b3765dff3e4bdeb8
    vcpkg-scripts version: 203f5f3be 2022-06-06 (12 hours ago)

[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/8852444/install-x64-windows-dbg-out.log)
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/8852445/install-x86-windows-dbg-out.log)

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

### PR DESCRIPTION
**Pull request by Adela0814**
Fix #25109 

Added patch `fix-format-conflict.patch`(https://github.com/fmtlib/fmt/issues/2896), use qualified format_to call to avoid ADL conflict with `std::format_to`.

No feature needs to test.

### CODE REVIEW

**Comment by Osyotr** [comment]
This looks wrong. The patch only modifies docs (`doc/api.rst`).

**Comment by Adela0814** [comment]
Waiting for user verification.

**Comment by Adela0814** [comment]
> This looks wrong. The patch only modifies docs (`doc/api.rst`).

This change comes from upstream, `format_to` conflict with `std::format_to`, so add a namespace to avoid this.

**Comment by Neumann-A** [comment]
@Adela0814:

I think you have the wrong patch. vcpkg shouldn't install docs so there is no reason to patch docs.  

> This change comes from upstream, format_to conflict with std::format_to, so add a namespace to avoid this.

I know about that and encountered that in HPX. `*.rst` files are not source files so your current change does not lead to a different output library. Basically `FMT_DOC` should be set to false/off in the `portfile` which it is https://github.com/microsoft/vcpkg/blob/be9fb7d37753f0f5e8829a417a9786b1a0626dd7/ports/fmt/portfile.cmake#L18
which thus deactivates the subfolder `doc/` completely
https://github.com/fmtlib/fmt/blob/b135f1c01449f6f41be3933437797e768b54295a/CMakeLists.txt#L353-L355

So your change here doesn't do anything to the output of vcpkg.

**Comment by Neumann-A** [comment]
You want to patch `include/fmt/format-inl.h(78)`

**Comment by Adela0814** [comment]
> @Adela0814:
> 
> I think you have the wrong patch. vcpkg shouldn't install docs so there is no reason to patch docs.
> 
> > This change comes from upstream, format_to conflict with std::format_to, so add a namespace to avoid this.
> 
> I know about that and encountered that in HPX. `*.rst` files are not source files so your current change does not lead to a different output library. Basically `FMT_DOC` should be set to false/off in the `portfile` which it is
> 
> https://github.com/microsoft/vcpkg/blob/be9fb7d37753f0f5e8829a417a9786b1a0626dd7/ports/fmt/portfile.cmake#L18
> 
> 
> which thus deactivates the subfolder `doc/` completely
> https://github.com/fmtlib/fmt/blob/b135f1c01449f6f41be3933437797e768b54295a/CMakeLists.txt#L353-L355
> So your change here doesn't do anything to the output of vcpkg.

Thanks for your comment, I will update this PR.

**Comment by JonLiu1993** [comment]
@frits-greuter, Could you try if this PR solves your problem?

**Review by JonLiu1993 — APPROVED** [review]

File: ports/fmt/fix-format-conflict.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+   auto it = buffer_appender<char>(out);
+   if (message.size() <= inline_buffer_size - error_code_size)
+-    format_to(it, FMT_STRING("{}{}"), message, SEP);
+-  format_to(it, FMT_STRING("{}{}"), ERROR_STR, error_code);
++    fmt::format_to(it, FMT_STRING("{}{}"), message, SEP);
++  fmt::format_to(it, FMT_STRING("{}{}"), ERROR_STR, error_code);
+   FMT_ASSERT(out.size() <= inline_buffer_size, "");
+ }
+ 
+diff --git a/src/os.cc b/src/os.cc
```
Comment (Adela0814) [inline ports/fmt/fix-format-conflict.patch:16]: See https://github.com/fmtlib/fmt/pull/2922/files

**Pull request merged**